### DISCUSSION TITLE
How to add Coupler stops to main panel (Friesach)?

### DISCUSSION THREAD

**Discussion by jason-barba** [GrandOrgue/grandorgue#1924] [Help] [answered]
**Goal**
Display the Friesach stops "3. Schwellwerk U.O." and "4. Solowerk 8" from panel Coupler 3. Schwellwerk on the Friesach panel. 

**Question**
I know how to add pistons/labels etc on the Friesach panel, but I need guidance on how to add the above stops. 
Can I use a similar config as this example for this purpose?

[SetterElement009]
Type=GC
PositionX=1100
PositionY=140
DisplayAsPiston=Y
DispImageNum=2
DispLabelText=GC
DispLabelFontSize=6
DispLabelColour=BLACK
 
If so, what is the Type value for these two stops?

Any help is highly appreciated.

**My setup**
GO v3.12.1-1
Sampleset Friesach
Keyboard with 2 manuals
Foot controller FCB 1010

//Jens

**Comment by larspalo** [comment]
No, it's not working like that. The couplers you mention is coming from the GO built in coupler panels. They are not part of the sample set as such, nor are they available as setter elements. So, to use them as if in the organ file itself (and display on a sample set panel) you'll need to create/add the two couplers first for the Schwellwerk in the odf, then you can display them on a panel by modifying the odf appropriately.

As this will alter/change the original organ, I'd suggest saving it with a modified file name and ChurchName, but you do whatever you want.

**Reply by jason-barba** [reply]
OK. Can you please direct me towards an example. I do not understand all of the Coupler object description, I am sorry to say.

**Reply by larspalo** [reply, marked as answer]
Honestly, I think you'd be much better off using either GoOdf or OdfEdit when you try to modify an .organ file than trying to learn the pure text mechanics behind it. If you really want to learn the hardcore text editing necessary to make manual .organ file editing you should spend much more time to learn how the format actually works - and I'm not advocating it, you do so at your own risk and effort.

The GO version of Friesach unfortunately uses the old panel format which further complicates things as the new format is much easier to use with greater flexibility. From what I can see in the .organ file the [Coupler006] already is the same as you want for the normal coupler to the Solowerk 8' from the Schwellwerk, and it's already displayed (by means of a [Switch029] displayed as drawstop nbr 49, lowest and furthest to the right). To add the new unison off coupler to [Manual002] you need to increase the number of couplers there and reference the new coupler section, as well as adding that new coupler section like:

[Coupler007]
Name=Sw U.O.
DefaultToEngaged=N
UnisonOff=Y

as well as the display related lines, including Displayed=Y, or possibly like the other couplers instead reference a new displayed switch that will activate it and then the DeafultToEngaged should be left out for the coupler... I leave that excersise for you to complete, but as you likely want those two couplers to be used together as a way to only play the solowerk from the schwellwerk it might be useful to create a switch that will activate both couplers at once and never display the unison off coupler on its own - but that's up to you.

**Reply by jason-barba** [reply]
Thank your for the kind lecture :-) And Stop 49 I have missed completely.
I will look into the matter it further. Mayby the Stop 49 fulfils my requirement well enough.
At the moment I can control both coupler 3 stops (8 and U.O.) from the FCB1010, but I would like to have indications of their status on the main panel.

**Reply by eturpault** [reply]
> The GO version of Friesach unfortunately uses the old panel format which further complicates things as the new format is much easier to use with greater flexibility. 

I could add in OdfEdit a feature permitting to convert an ODF from old panel format to new panel format. Would anyone be interested in that?

**Reply by jason-barba** [reply]
I for one is interested.

**Reply by larspalo** [reply]
@eturpault @jason-barba GoOdf "converts" any opened .organ file to new panel format automatically, but it also does some other efficiency related optimizations.
